File: Construct3/Construct3/Library/Telerik.OpenAccess.SDK/IDEIntegrations/templates/newWinFormNavigator.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms

Partial Public Class %ClassName%
    Inherits Form
    ''' <summary>
    ''' Summary description for %ClassName%.
    ''' </summary>
    Public Sub New()
        InitializeComponent()
        AddHandler Me.Closed, AddressOf FormClosedEvent
        ' Add the following code if you want to dispose on close
        Me.bindingNavigator1.BindingSource = objectView1
    End Sub

    Private Sub FormClosedEvent(ByVal sender As Object, ByVal e As System.EventArgs)
        Me.Dispose()
    End Sub

    Private Sub bindingNavigatorAddNewItem_Click(ByVal sender As Object, ByVal e As EventArgs)
        Try
            ' Clear out the current edits
            Me.BindingContext(objectView1).EndCurrentEdit()
            Me.BindingContext(objectView1).AddNew()
        Catch eEndEdit As System.Exception
            System.Windows.Forms.MessageBox.Show(eEndEdit.Message)
        End Try

    End Sub



    Private Sub bindingNavigatorSaveItem_Click(ByVal sender As Object, ByVal e As EventArgs)
        Me.BindingContext(objectView1).EndCurrentEdit()
        Me.objectProvider1.SaveAll()

    End Sub

    Private Sub bindingNavigatorDeleteItem_Click(ByVal sender As Object, ByVal e As EventArgs)

    End Sub

    Private Sub bindingNavigatorCancelItem_Click(ByVal sender As Object, ByVal e As EventArgs)
        Dim r As DialogResult = MessageBox.Show("Do you really want to cancel ?", "confirm", MessageBoxButtons.YesNo)
        If r = DialogResult.Yes Then
            Me.BindingContext(objectView1).CancelCurrentEdit()
            Me.objectProvider1.CancelAll()
        End If
    End Sub

End Class

File: Construct3/Construct3/Library/Telerik.OpenAccess.SDK/IDEIntegrations/templates/newWinFormNavigator.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace %DefaultNamespace%
{
    public partial class %ClassName% : Form
    {
        /// <summary>
	    /// Summary description for %ClassName%.
	    /// </summary>
        public %ClassName%()
        {
            InitializeComponent();
            // Add the following code if you want to dispose on close
            this.Closed += new System.EventHandler(this.FormClosedEvent);            
            this.bindingNavigator1.BindingSource = objectView1;            
        }

        private void FormClosedEvent(object sender, System.EventArgs e)
		{
			this.Dispose();
		}

        private void bindingNavigatorAddNewItem_Click(object sender, EventArgs e)
        {
            try
            {
                // Clear out the current edits
                this.BindingContext[objectView1].EndCurrentEdit();
                this.BindingContext[objectView1].AddNew();
            }
            catch (System.Exception eEndEdit)
            {
                System.Windows.Forms.MessageBox.Show(eEndEdit.Message);
            }
            
        }

                    

        private void bindingNavigatorSaveItem_Click(object sender, EventArgs e)
        {
            this.BindingContext[objectView1].EndCurrentEdit();
            this.objectProvider1.SaveAll();
            
        }

        private void bindingNavigatorCancelItem_Click(object sender, EventArgs e)
        {
            DialogResult r = MessageBox.Show("Do you really want to cancel ?", "confirm", MessageBoxButtons.YesNo);
            if (r == DialogResult.Yes)
            {
                this.BindingContext[objectView1].CancelCurrentEdit();
                this.objectProvider1.CancelAll();
                
            }
        }

    }
}